RestartManager.js

```diff
@@ -0,0 +1,39 @@
+let NativeCodePush = require("react-native").NativeModules.CodePush;
+
+const RestartManager = (() => {
+    let _allowed = true;
+    let restartPending = false;
+
+    function tryRestart() {
+        if (restartPending && _allowed == true) {
+            NativeCodePush.restartApp(true);
```

Comment:
By pasing `true` to `restartApp`, you're already preventing the restart from happening unless there is a pending update. Do we need the `restartPending` flag at that point?